src/state/trajectory/actions.ts

```diff
@@ -196,3 +195,36 @@ export function cancelAutoconversion(): CancelConversionAction {
         type: CANCEL_CONVERSION,
     };
 }
+
+export function setUserSelectedUIData(
+    payload: UIDisplayData
+): SetUserSelectedUIDataAction {
+    return {
+        payload,
+        type: SET_USER_SELECTED_UI_DATA,
+    };
+}
+
+export function setDefaultUIData(
+    payload: UIDisplayData
+): SetDefaultUIDataAction {
+    return {
+        payload,
+        type: SET_DEFAULT_UI_DATA,
+    };
+}
+
+export function setCurrentColorSettings(payload: {
+    currentColorSettings: ColorSettings;
+}): SetCurrentColorSettingsAction {
+    return {
+        payload,
+        type: SET_CURRENT_COLOR_SETTINGS,
+    };
+}
```

Comment:
When dispatching actions from inside redux logics I have found it necessary to wrap payloads in an object rather than apply them directly, to avoid a type error, the `dispatch` call rejects the type of the raw payload.

So in this case `payload: ColorSettings` threw an error but `{currentColorSettings: ColorSettings}` did not.

I don't love that this is not obvious when reading this file and looking at the different actions and how their payloads are typed, but couldn't find a way to make it work without.